torch/autograd/_functions/linalg.py

```diff
@@ -102,3 +102,42 @@ def backward(ctx, grad_output, grad_LU=None):
         grad_b, _ = torch.gesv(grad_output, a.t())
         grad_a = -torch.mm(grad_b, X.t())
         return grad_b, grad_a
+
+class Symeig(Function):
+
+    @staticmethod
+    def forward(ctx, input, eigenvectors=False, upper=True):
+        ctx.eigenvectors = eigenvectors
```

Comment:
no, i dont use it elsewhere atm - it's there for your second point on what to do if `eigenvectors=False`. Seems like you shouldn't be able to calculate the backward if so. Should I raise an exception in `backward` if it's called with `eigenvectors=False` ?